core/src/wsv.rs

```diff
@@ -276,7 +276,9 @@ impl WorldStateView {
 
     /// Send [`Event`]s to known subscribers.
     fn produce_event(&self, event: impl Into<Event>) {
-        let _result = self.events_sender.send(event.into());
+        if let Err(error) = self.events_sender.send(event.into()) {
+            debug!(%error, event = ?error.0, "Failed send event");
+        }
```

Comment:
the pattern where you send event and log an error if it occurs 4 times in this PR and possibly in other places. You think you could extract this functionality into some `produce_event` function or wrap the sender so that it's `send` method logs on fail?

On the other hand, I also find it good as is

Reply:
Good point